source/MRMesh/MRObject.cpp

```diff
@@ -251,12 +251,14 @@ bool Object::addChild( std::shared_ptr<Object> child, bool recognizedChild )
     child->parent_ = this;
     if ( recognizedChild )
     {
+        child->propagateWorldXfChangedSignal_();
```

Comment:
propagateSignal must be called after children_ update, otherwise the state of Object tree is invalid during propagate